frontend/public/components/storage-class-form.tsx

```diff
@@ -501,25 +551,48 @@ export class StorageClassForm_ extends React.Component<
         },
       },
     },
-  });
+  };
 
   reclaimPolicies = {
     Retain: 'Retain',
     Delete: 'Delete',
   };
 
+  csiProvisionerMap = (csiData) => {
+    const csiListedProvisioner: string[] = _.keys(this.CSIStorageTypes);
+    csiData.map((csi) => {
+      csiListedProvisioner.some((provisioner) => {
+        const hasProvisioner = getName(csi).includes(provisioner);
+        if (hasProvisioner) {
+          const provisionerData = _.cloneDeep(this.CSIStorageTypes[provisioner]);
+          provisionerData.provisioner = getName(csi);
+          this.storageTypes[getName(csi)] = provisionerData;
```

Comment:
It's unexpected that calling `some` would have a side effect.

Reply:
I'd think you would want to have `storageTypes` in state since the list is no longer status and it's something that is used in render.

Reply:
I declared storageTypes as another instance and freezed the other parameter objects.